source/base/data_out_base.cc

```diff
@@ -7741,19 +7746,58 @@ DataOutInterface<dim, spacedim>::write_vtu_in_parallel(
                                   vtk_flags,
                                   ss);
 
-    ierr = MPI_File_write_ordered(
-      fh, ss.str().c_str(), ss.str().size(), MPI_CHAR, MPI_STATUS_IGNORE);
+    // using prefix sum to find specific offset to write at.
+    const std::uint64_t size_on_proc = ss.str().size();
+    std::uint64_t prefix_sum = 0;
+    ierr =
+      MPI_Exscan(&size_on_proc, &prefix_sum, 1, MPI_UINT64_T, MPI_SUM, comm);
+    AssertThrowMPI(ierr);
+
+    // Locate specific offset for each processor.
+    const MPI_Offset offset = static_cast<MPI_Offset>(header_size) + prefix_sum;
+
+    ierr =
+      Utilities::MPI::LargeCount::MPI_File_write_at_all_c(fh,
+                                                          offset,
+                                                          ss.str().c_str(),
+                                                          ss.str().size(),
+                                                          MPI_CHAR,
+                                                          MPI_STATUS_IGNORE);
     AssertThrowMPI(ierr);
+
+    // Sending the offset for writing the footer to rank 0.
+    if (myrank == n_ranks - 1)
+      {
+        const std::uint64_t footer_offset = size_on_proc + offset;
+        ierr = MPI_Send(&footer_offset, 1, MPI_UINT64_T, 0, 0, comm);
+        AssertThrowMPI(ierr);
+      }
   }
 
   // write footer
   if (myrank == 0)
     {
       std::stringstream ss;
       DataOutBase::write_vtu_footer(ss);
-      unsigned int footer_size = ss.str().size();
-      ierr = MPI_File_write_shared(
-        fh, ss.str().c_str(), footer_size, MPI_CHAR, MPI_STATUS_IGNORE);
+      const unsigned int footer_size = ss.str().size();
+
+      std::uint64_t footer_offset;
+      ierr = MPI_Recv(&footer_offset,
+                      1,
+                      MPI_UINT64_T,
+                      n_ranks - 1,
+                      0,
+                      comm,
+                      MPI_STATUS_IGNORE);
+      AssertThrowMPI(ierr);
+
+      // Writing Footer.
+      ierr = MPI_File_write_at(fh,
+                               footer_offset,
+                               ss.str().c_str(),
+                               footer_size,
+                               MPI_CHAR,
+                               MPI_STATUS_IGNORE);
```

Comment:
Is this the right function call, or don't you need one from the big_mpi library?

Reply:
To the best of my knowledge, the header and footer are very small in terms of size, so using write_at and write_at_c (The one in big_mpi library) will not make a difference :)

Reply:
I agree with Pengfei.

Reply:
Could we do it just out of consistency and so that people like me don't have to think about the question in the future :-)

Reply:
Sure!